src/test/EC3ConfigurationsClient.test.ts

```diff
@@ -0,0 +1,229 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import * as chaiAsPromised from "chai-as-promised";
+import { expect, use } from "chai";
+import * as sinon from "sinon";
+import { EC3ConfigurationsClient } from "../reporting/clients/EC3ConfigurationsClient";
+import { EC3Configuration, EC3ConfigurationCreate, EC3ConfigurationUpdate } from "../reporting/interfaces/EC3Configurations";
+use(chaiAsPromised);
+
+describe("EC3ConfigurationsClient", () => {
+  const configurationsClient: EC3ConfigurationsClient = new EC3ConfigurationsClient();
+  let fetchStub: sinon.SinonStub;
+  let requestStub: sinon.SinonStub;
+
+  beforeEach(() => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    fetchStub = sinon.stub(configurationsClient, "fetchJSON" as any);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    requestStub = sinon.stub(configurationsClient, "createRequest" as any);
+    requestStub.returns("pass");
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it("EC3ConfigurationsClient - change base path", async () => {
+    const client = new EC3ConfigurationsClient("BASE");
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    fetchStub = sinon.stub(client, "fetchJSON" as any);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    requestStub = sinon.stub(client, "createRequest" as any);
+
+    const returns = {
+      configuration: {
+        id: 1,
+      },
+    };
+    fetchStub.resolves(returns);
+    await client.getConfiguration("auth", "configurationId");
+    expect(fetchStub.getCall(0).args[0]).to.match(new RegExp("^BASE"));
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith

I'd prefer not to use regex if it is possible.

Reply:
comparing to substring